staging/src/k8s.io/apiserver/pkg/storage/storagebackend/factory/etcd3.go

```diff
@@ -512,57 +516,51 @@ func startDBSizeMonitorPerEndpoint(client *clientv3.Client, interval time.Durati
 	}, nil
 }
 
-// checkVersion checks for deprecated versions of etcd and returns an error if find any
-// currently the versions < 3.4.25, between 3.5.0 and 3.5.7
-// based on this: https://github.com/etcd-io/etcd/issues/15220
+// checkVersion checks for deprecated versions of etcd that are affected by known issues.
+// It returns an error if the cluster is running a deprecated version, specifically versions below 3.4.25
+// or between 3.5.0 and 3.5.7. For more details on this issue, refer to:
+//   - https://github.com/etcd-io/etcd/issues/15220
+//   - https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/2340-Consistent-reads-from-cache/README.md#bug-in-etcd-progress-notification
 func checkVersion(c *clientv3.Client, dialTimeOut time.Duration) (err error) {
 	eps := c.Endpoints()
-	baseCtx := context.TODO()
+	ctx := context.TODO()
 
-	ctx, cancel := context.WithCancel(baseCtx)
 	if dialTimeOut > 0 {
-		cancel()
-		ctx, cancel = context.WithTimeout(baseCtx, dialTimeout)
-		defer cancel()
+		// errgroup will handle closing context when it exits
+		ctx, _ = context.WithTimeout(ctx, dialTimeout)
```

Comment:
```suggestion
		var cancel context.CancelFunc
		ctx, cancel = context.WithTimeout(ctx, dialTimeout)
		defer cancel()
```
You still need to cancel this function :P

Reply:
That was my concern too :D, i put a comment there just for that, thanks for clarification